dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/proxy/AbstractProxyInvoker.java

```diff
@@ -97,6 +99,11 @@ public Result invoke(Invocation invocation) throws RpcException {
                 }
             }
 
+            Map<String, String> invocationAttachments = invocation.getAttachments();
+            if (CollectionUtils.isNotEmptyMap(invocationAttachments)) {
+                RpcContext.getClientAttachment().getObjectAttachments().putAll(invocationAttachments);
+            }
+
```

Comment:
It would be better to replace the way to get attachment in customized filter of rest protocol to  `org.apache.dubbo.rpc.RpcServiceContext#getInvocation` to get the origin invocation.

Reply:
Why does the AbstractProxyInvoker on the provider side use ClientAttachment?